packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -195,18 +184,41 @@ contract BridgePool is Testable {
         _emitDepositRelayedEvent(depositData, keccak256(getRelayAncillaryData(depositData, relayData)), depositHash);
     }
 
-    // TODO: Implement fully the following functions:
-    function speedUpRelay(bytes32 _depositHash) public {
-        RelayData storage relay = relays[_depositHash];
-        require(relays[_depositHash].relayState == RelayState.Pending, "Can only speed up pending slow relay");
-        require(relays[_depositHash].instantRelayer == address(0), "Relay has already been instant relayed");
+    /**
+     * @notice Instantly relay a deposit amount minus fees. Instant relayer earns a reward following the pending relay
+     * challenge period.
+     * @dev Caller must have approved this contract to spend the deposit amount of L1 tokens to relay. There can only
+     * be one instant relayer per relay attempt and disputed relays cannot be sped up.
+     * @param _depositData Unique set of L2 deposit data that caller is trying to instantly relay.
+     */
+    function speedUpRelay(DepositData memory _depositData) public {
+        bytes32 depositHash = _getDepositHash(_depositData);
+        RelayData storage relay = relays[depositHash];
+        require(relays[depositHash].relayState == RelayState.Pending, "Can only speed up pending slow relay");
+        require(relays[depositHash].instantRelayer == address(0), "Relay has already been instant relayed");
         relay.instantRelayer = msg.sender;
-        // TODO: Pull funds from caller.
-        emit RelaySpedUp(_depositHash, msg.sender);
+
+        // Pull relay amount minus fees from caller.
+        IERC20(_depositData.l1Token).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _depositData.amount - _getRealizedFeeAmount(relay.realizedFeePct, _depositData.amount)
```

Comment:
nit: put this on a separate line for readability

Reply:
I'm trying to keep the stack small and use fewer intermediate variables. Do you think this gives gas savings either way?